### PR TITLE
IP Quality Score changes

### PR DESCRIPTION
**Pull request by krishnasism**

### CODE REVIEW

File: modules/sfp_ipqualityscore.py [hunk 1]
```diff
@@ -156,21 +160,74 @@ def handleEvent(self, event):
...
         recentAbuse = data.get('recent_abuse')
         botStatus = data.get('bot_status')
+        malicious = False
+        maliciousDesc = ""
+
         if fraudScore >= self.opts['abuse_score_threshold'] or recentAbuse or botStatus:
             evt = SpiderFootEvent("RAW_RIR_DATA", str(data), self.__name__, event)
             self.notifyListeners(evt)
+            malicious = True
+            maliciousDesc = f"IPQualityScore [ {eventData} ]\n"
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:170]: Should be no spaces between the `[` `]` and `eventData`.

File: modules/sfp_ipqualityscore.py [hunk 2]
```diff
@@ -156,21 +160,74 @@ def handleEvent(self, event):
...
+            if city:
+                geoInfo += city + ", "
+            if region:
+                geoInfo += region + ", "
+            if country:
+                geoInfo += country + " "
+            if zipcode:
+                geoInfo += zipcode + ". "
+
+            evt = SpiderFootEvent("GEOINFO", geoInfo, self.__name__, event)
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:193]: `GEOINFO` isn't declared in `producedEvents()`

File: modules/sfp_ipqualityscore.py [hunk 3]
```diff
@@ -156,21 +160,74 @@ def handleEvent(self, event):
...
+            region = data.get('region')
+
+            if city:
+                geoInfo += city + ", "
+            if region:
+                geoInfo += region + ", "
+            if country:
+                geoInfo += country + " "
+            if zipcode:
+                geoInfo += zipcode + ". "
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:191]: I don't think you need a `.`.

File: modules/sfp_ipqualityscore.py (hunk 2 above)
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:193]: `geoinfo` may still be empty by this stage, so you should wrap this in a `if geoInfo` clause.

File: modules/sfp_ipqualityscore.py [hunk 4]
```diff
@@ -156,21 +160,74 @@ def handleEvent(self, event):
...
-                evt = SpiderFootEvent("MALICIOUS_EMAILADDR", "IPQualityScore [" + eventData + "]", self.__name__, event)
+
+            if data.get('disposable'):
+                evt = SpiderFootEvent("EMAILADDR_DISPOSABLE", eventData, self.__name__, event)
                 self.notifyListeners(evt)
-            elif eventName == "IP_ADDRESS":
-                evt = SpiderFootEvent("MALICIOUS_IPADDR", "IPQualityScore [" + eventData + "]", self.__name__, event)
+
+            if data.get('leaked'):
+                evt = SpiderFootEvent("EMAILADDR_COMPROMISED", f"{eventData} [ {data.get('suggested_domain')} ]", self.__name__, event)
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:207]: Please remove paces after `[` and before `]`.

File: modules/sfp_ipqualityscore.py [hunk 5]
```diff
@@ -156,21 +160,74 @@ def handleEvent(self, event):
...
+
+            if city:
+                geoInfo += city + ", "
+            if region:
+                geoInfo += region + ", "
+            if country:
+                geoInfo += country + " "
+
+            evt = SpiderFootEvent("GEOINFO", geoInfo, self.__name__, event)
+            self.notifyListeners(evt)
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:231]: Since this code for geo data is duplicated, maybe move it to its own function?

Comment (smicallef) [inline modules/sfp_ipqualityscore.py:231]: 3045835950

File: modules/sfp_ipqualityscore.py [hunk 6]
```diff
@@ -156,21 +160,74 @@ def handleEvent(self, event):
...
+        malicious = False
+        maliciousDesc = ""
+
         if fraudScore >= self.opts['abuse_score_threshold'] or recentAbuse or botStatus:
             evt = SpiderFootEvent("RAW_RIR_DATA", str(data), self.__name__, event)
             self.notifyListeners(evt)
+            malicious = True
+            maliciousDesc = f"IPQualityScore [ {eventData} ]\n"
+
+        if eventName == "PHONE_NUMBER":
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:172]: I believe the API also indicates the phone number type, or at least if it's disposable? That way this module could return `PHONE_NUMBER_TYPE`, and we could also add `PHONE_NUMBER_DISPOSABLE` as a type, just as we have for emails.

Comment (krishnasism) [reply modules/sfp_ipqualityscore.py:172]: Don't see anything like this in the response

File: modules/sfp_ipqualityscore.py [hunk 7]
```diff
@@ -156,21 +183,58 @@ def handleEvent(self, event):
...
-                evt = SpiderFootEvent("MALICIOUS_EMAILADDR", "IPQualityScore [" + eventData + "]", self.__name__, event)
+
+            if data.get('disposable'):
+                evt = SpiderFootEvent("EMAILADDR_DISPOSABLE", eventData, self.__name__, event)
                 self.notifyListeners(evt)
-            elif eventName == "IP_ADDRESS":
-                evt = SpiderFootEvent("MALICIOUS_IPADDR", "IPQualityScore [" + eventData + "]", self.__name__, event)
+
+            if data.get('leaked'):
+                evt = SpiderFootEvent("EMAILADDR_COMPROMISED", f"{eventData} [{data.get('suggested_domain')}]", self.__name__, event)
```
Comment (smicallef) [inline modules/sfp_ipqualityscore.py:222]: Why `suggested_domain`?

**Pull request merged**